### PR TITLE
haxe: add 4.1 and 4.0 compiler versions

### LINKED ISSUE #95638: Update Haxe Version to 4.1.3
There was a new release on Haxe not too long ago. It's now on 4.1.3 

https://github.com/HaxeFoundation/haxe/releases/tag/4.1.3

The version in nixpkgs is pretty old, it's still on 3.4.6

Would be cool if this could be updated.

Maintainer: @MarcWeber

### PR DESCRIPTION
**Pull request by sternenseemann**
###### Motivation for this change

Fixes #95638. cc @OPNA2608 

Arguably the extlib 1.7.7 override is not ideal, but I don't really want to be stuck with it in `ocamlPackages` as long as we don't need it for anything else. The compatibility patch from 4.2 unfortunately doesn't even apply on 4.1.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 118112` at 108cd8e8 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>haxe_4_0</li>
    <li>haxe_4_1</li>
  </ul>
</details>

**Comment by OPNA2608** [comment]
Sorry for the late response! Tested `haxe_4_1` and `haxe_4_0` with a game project, both worked fine.

File: pkgs/development/compilers/haxe/default.nix [hunk 1]
```diff
@@ -1,9 +1,23 @@
...
 let
+  defaultOCamlPackages = ocaml-ng.ocamlPackages.overrideScope'
+    (self: super: {
+      # haxe 4.0 and 4.1 unfortunately need an older version of extlib
+      ocaml_extlib-1-7-7 = super.ocaml_extlib.overrideAttrs (_: rec {
+        version = "1.7.7";
+        src = fetchurl {
+          url = "https://github.com/ygrek/ocaml-extlib/releases/download/${version}/extlib-${version}.tar.gz";
+          sha256 = "1sxmzc1mx3kg62j8kbk0dxkx8mkf1rn70h542cjzrziflznap0s1";
+        };
```
Comment (OPNA2608) [inline pkgs/development/compilers/haxe/default.nix:12]: You could reuse `fetchFromGitHub` here.

**Review by OPNA2608 — CHANGES_REQUESTED** [review]
Tiny nitpick, otherwise looks good IMO.

File: pkgs/development/compilers/haxe/default.nix (hunk 1 above)
Comment (sternenseemann) [reply pkgs/development/compilers/haxe/default.nix:12]: No, that is a release asset and not the automatically generated archive, so we need to use `fetchurl`.

Comment (OPNA2608) [reply pkgs/development/compilers/haxe/default.nix:12]: Oh I see, nevermind then.

**Review by OPNA2608 — APPROVED** [review]

**Comment by sternenseemann** [comment]
@ofborg build haxe_4_0 haxe_4_1 ocamlPackages.ocaml_extlib ocamlPackages.ocaml_extlib-1-7-7

**Pull request merged**